lib/cdo/school_autocomplete.rb

```diff
@@ -0,0 +1,35 @@
+require 'singleton'
+require 'triez'
+
+class SchoolAutocomplete
+  include Singleton
+
+  def initialize
+    @schools = Triez.new value_type: :object
+    School.all.each do |school|
+      key = "#{school[:name]} #{school[:city]} #{school[:zip]}".downcase
+      @schools.change_all(:suffix, key) {school}
+    end
+  end
+
+  def get_matches(query, limit)
+    results = []
+    if query.length >= 3
+      @schools.search_with_prefix(query.downcase, limit: limit) do |_, school|
+        results << Serializer.new(school).attributes
+      end
+    end
+    return results
+  end
+
+  class Serializer < ActiveModel::Serializer
+    attributes :id, :name, :city, :state, :zip, :school_district_id
+    def name
+      object.name.titleize
+    end
+    def city
+      object.city.titleize
+    end
+  end
```

Comment:
Out of curiosity, why is this `ActiveModel::Serializer` in lib, instead of the usual (Rails preferred) location `dashboard/app/serializers`?

Along these lines, we tend to favor classes in their own files, and to name the serializer more specifically, e.g. file: `dashboard/app/serializers/school_serializer.rb`, class: 'SchoolSerializer`